tiledb/sm/stats/duration_instrument.h

```diff
@@ -27,79 +27,64 @@
  *
  * @section DESCRIPTION
  *
- * This file contains the TimerStat definition.
- *
- * This allows starting and stopping concurrent timers on different threads,
- * but does not count overlapping time periods toward the recorded duration.
- *
- * Thread-safety: none.
+ * This file defines class DurationInstrument.
  */
 
-#ifndef TILEDB_TIMER_STAT_H
-#define TILEDB_TIMER_STAT_H
+#ifndef TILEDB_DURATION_INSTRUMENT_H
+#define TILEDB_DURATION_INSTRUMENT_H
 
-#include <thread>
-#include <unordered_map>
+#include "tiledb/common/common.h"
+
+using namespace tiledb::common;
 
 namespace tiledb {
 namespace sm {
 namespace stats {
 
-class TimerStat {
+class Stats;
+
+/**
+ * This class contains a simple duration instrument.
+ */
+class DurationInstrument {
  public:
   /* ****************************** */
   /*   CONSTRUCTORS & DESTRUCTORS   */
   /* ****************************** */
 
-  /** Constructor. */
-  TimerStat();
+  /** Constructs an empty duration instrument object. */
+  DurationInstrument()
+      : parent_stats_(nullptr)
+      , stat_name_("") {
+  }
 
-  /** Destructor. */
-  ~TimerStat() = default;
+  /** Constructs a duration instrument object. */
+  DurationInstrument(Stats* parent_stats, const std::string stat_name)
+      : parent_stats_(parent_stats)
```

Comment:
Is there an implicit requirement here that `parents_stats` should never be null? If so, it should be checked in the body.